pages/api/augment-request.ts

```diff
@@ -0,0 +1,145 @@
+  return `https://oracle.umaproject.org/request?transactionHash=${txHash}&chainId=${chainId}&oracleType=${oracleType}`;
+}
+
+async function constructOptimisticOraclesByChain(chainId: number): Promise<(Contract | null)[]> {
+  const optimisticOracle = await constructContractOnChain(chainId, "OptimisticOracle");
+  const optimisticOracleV2 = await constructContractOnChain(chainId, "OptimisticOracleV2");
+
+  // Only mainnet has a SkinnyOptimisticOracle so only construct it here. All other chains of interest have OOv1 and V2.
+  const skinnyOptimisticOracle =
+    chainId == 1 ? await constructContractOnChain(chainId, "SkinnyOptimisticOracle") : null;
+  return [optimisticOracle, optimisticOracleV2, skinnyOptimisticOracle];
+}
+
+async function getOptimisticOraclePriceRequestEventsByChainId(chainId: number): Promise<any[]> {
+  const chainOptimisticOracles = await constructOptimisticOraclesByChain(chainId);
+
+  const requests = await Promise.all(
+    chainOptimisticOracles.map((optimisticOracle) =>
+      optimisticOracle ? optimisticOracle.queryFilter(optimisticOracle.filters.RequestPrice()) : null
+    )
+  );
+
+  return requests
+    .map((oracleRequests, index) => {
+      return oracleRequests
+        ? oracleRequests.map((ooRequest) => {
+            return {
+              transactionHash: ooRequest.transactionHash,
+              identifier: ooRequest?.args?.identifier,
+              time: Number(ooRequest?.args?.timestamp),
+              oracleType: OptimisticOracleType[index],
+            };
+          })
+        : [];
+    })
+    .flat();
+}
+
+async function getRequestTxFromL1RequestInformation(
+  l1Requests: Request[]
+): Promise<{ requestTransactionHash: string; chainId: number; oracleType: any }[]> {
+  const ooChainIds = getOoChainIds();
+
+  const chainOoRequests = await Promise.all(
+    ooChainIds.map((chainId) => getOptimisticOraclePriceRequestEventsByChainId(chainId))
+  );
+  const ooRequestTxs: { requestTransactionHash: string; chainId: number; oracleType: any }[] = [];
+  l1Requests.forEach((l1Request) => {
+    let foundOoTx = { requestTransactionHash: "", chainId: 0, oracleType: "" };
+    chainOoRequests.forEach((_, ooChainId) => {
+      chainOoRequests[ooChainId].forEach((ooRequest, index) => {
+        if (
+          l1Request.identifier.toLowerCase() == ooRequest.identifier.toLowerCase() &&
+          l1Request.time == ooRequest.time
+        ) {
+          foundOoTx = {
+            requestTransactionHash: ooRequest.transactionHash,
+            chainId: ooChainIds[ooChainId],
+            oracleType: ooRequest.oracleType,
+          };
+          return;
+        }
+      });
+    });
+    ooRequestTxs.push(foundOoTx);
+  });
+  return ooRequestTxs;
+}
+
+async function getAugmentingRequestInformation(l1Requests: Request[]) {
+  const votingV1 = await constructContractOnChain(1, "Voting");
+  //todo: add voting v2 when released.
```

Comment:
lol I can't believe this, but `//todo` with no space between the `//` and `todo` causes a build error. Add a space to fix.

Reply:
thats good! this is a common linting package restriction we have in many other repos as well.